### PR TITLE
f-contact-preferences@0.11.0 - Add info and error logging

### PR DESCRIPTION
**Pull request by xander-marjoram**
### Added
- `info` and `error` logs to preferences fetch and save requests
- Unit tests

### Removed
- Unused variable from story helper (integrated changes from #1543)

---

## UI Review Checks
- [ ] README and/or UI Documentation has been [created|updated]
- [x] Unit tests have been [created|updated]
- [ ] This code has been checked with regard to [our accessibility standards](http://fozzie.just-eat.com/documentation/general/accessibility/checklist)

## Browsers Tested

- [x] Chrome (latest)
- [ ] Internet Explorer 11
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### List any other browsers that this PR has been tested in:

- [View the Browser Support Checklist](http://fozzie.just-eat.com/documentation/general/browser-support)

### CODE REVIEW

File: packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js [hunk 1]
```diff
@@ -53,11 +57,9 @@ const mountContactPreferences = ({
...
     data = dataDefaults,
-    storeOverride = null,
     initialiseOverride = null
 } = {}) => {
-    const store = storeOverride || createStore({ state, actions });
-    store.registerModule = registerStoreModuleSpy;
+    const store = createStore({ state, actions });
 
     if (initialiseOverride) {
         ContactPreferences.methods.initialise = initialiseOverride;
```
Comment (AndyMayJE) [inline packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js]: Just spotted this in `f-account-info` once `ContactPreferences.methods.initialise` has been overridden it is never reset to the original method.

File: packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js [hunk 2]
```diff
@@ -100,24 +100,23 @@ describe('ContactPreferences Component', () => {
...
     describe('when creating the component', () => {
-        it('should register the Store Module', () => {
+        it('should register the Store Module', async () => {
             // Arrange & Act
-            wrapper = mountContactPreferences({ storeOverride: new Vuex.Store() });
+            wrapper = await mountContactPreferences();
 
             // Assert
-            expect(registerStoreModuleSpy).toHaveBeenCalled();
+            expect(wrapper.vm.$store.state.fContactPreferencesModule).toBeDefined();
```
Comment (oliversweb) [inline packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js:113]: Does this prove that **our** component registered the module?

File: packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js [hunk 3]
```diff
@@ -164,35 +163,57 @@ describe('ContactPreferences Component', () => {
...
             expect(wrapper.vm.isFormDirty).toEqual(false);
         });
 
-        it('should not call initialise method if the authorisation has not completed', () => {
+        it('should log an info log', async () => {
+            // Arrange & Act
+            wrapper = await mountContactPreferences();
+
+            // Assert
+            expect(logMocks.info).toHaveBeenCalledTimes(1);
```
Comment (oliversweb) [inline packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js:182]: Might be best to check we are sending the correct tags (ditto for the other log tests).

File: packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js (hunk 2 above)
Comment (xander-marjoram) [reply packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js:113]: No, good catch. Not sure why I removed it.

**Review by oliversweb — APPROVED** [review]

**Review by AndyMayJE — APPROVED** [review]

**Pull request merged**